core/src/main/java/org/elasticsearch/index/query/RegexpQueryBuilder.java

```diff
@@ -59,35 +156,47 @@ public RegexpQueryBuilder flags(RegexpFlag... flags) {
                 value |= flag.value;
             }
         }
-        this.flags = value;
+        this.flagsValue = value;
+        return this;
+    }
+
+    public RegexpQueryBuilder flags(int flags) {
+        this.flagsValue = flags;
         return this;
     }
 
+    public int flags() {
+        return this.flagsValue;
+    }
+
     /**
      * Sets the regexp maxDeterminizedStates.
      */
     public RegexpQueryBuilder maxDeterminizedStates(int value) {
         this.maxDeterminizedStates = value;
-        this.maxDetermizedStatesSet = true;
```

Comment:
If you ask me I'd be tempted to say that a bug is a bug and should definitely be backported. However, there are bugs which are only relevant in master, and so I thought that the query refactoring branch would be merged to master before 2.0 release. For this bug, I'll submit a fix for sure.